spot_wrapper/spot_dancing.py

```diff
@@ -0,0 +1,147 @@
+import sys
+import time
+import logging
+
+import bosdyn.client
+import bosdyn.client.util
+from bosdyn.choreography.client.choreography import (
+    ChoreographyClient,
+    load_choreography_sequence_from_txt_file,
+)
+from bosdyn.client import ResponseError
+from bosdyn.client.exceptions import UnauthenticatedError
+from bosdyn.client.lease import LeaseClient, LeaseKeepAlive
+from bosdyn.client.license import LicenseClient
+
+
+class SpotDancing:
+    def __init__(
+        self,
+        hostname: str,
+        robot_name: str,
+        username: str,
+        password: str,
+        logger: logging.Logger,
+    ):
+        self._hostname = hostname
+        self._robot_name = robot_name
+        self._username = username
+        self._password = password
+        self._logger = logger
+
+        sdk = bosdyn.client.create_standard_sdk("UploadChoreography")
+        sdk.register_service_client(ChoreographyClient)
+        self._robot = sdk.create_robot(self._hostname)
+        if not self._robot:
+            print("create robot failed!")
+        authenticated = self.authenticate(
+            self._robot, self._username, self._password, self._logger
+        )
+        if not authenticated:
+            print("Authentication failed!")
+            return
+        license_client = self._robot.ensure_client(LicenseClient.default_service_name)
+        if not license_client.get_feature_enabled([ChoreographyClient.license_name])[
+            ChoreographyClient.license_name
+        ]:
+            print("This robot is not licensed for choreography.")
+            sys.exit(1)
+        lease_client = self._robot.ensure_client(LeaseClient.default_service_name)
+        lease = lease_client.acquire()
+        lk = LeaseKeepAlive(lease_client)
+        self._choreography_client = self._robot.ensure_client(
+            ChoreographyClient.default_service_name
+        )
+
+    @staticmethod
+    def authenticate(robot, username, password, logger):
```

Comment:
Import the authenticate method from the wrapper rather than copying the code.

Reply:
Is there a reason this should authenticate separately and not just use the `robot` variable created in SpotWrapper similar to the modules in #6 ?

Reply:
Probably not. At the moment the changes indicate a standalone script, what I meant when suggesting modularity was to have it included in the wrapper initialisation like other modules in the PR you mentioned. It should probably print a short info message to indicate whether choreography is available or not.

Reply:
Agree about it being a standalone script at the moment. The question about using the `robot` variable was more directed at the author of the PR. @Guptabot